### PR TITLE
Drop Julia 0.6 Support

### PR DESCRIPTION
**Pull request by fchorney**
* Raise minimum Julia version to 0.7
* Replace 0.6 on CI with 1.1
* Remove dependency on Compat

### CODE REVIEW

**Review by iamed2 — APPROVED** [review]

**Comment by nicoleepp** [comment]
Also LGTM

File: src/FTPObject.jl [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
-const complete_transfer_code = 226
-
-
 mutable struct FTP
     ctxt::ConnContext
 
     function FTP(ctxt::ConnContext)
         ftp = new(ctxt)
-        @compat finalizer(close, ftp)  # 0.7.0-DEV.2562
+        finalizer(close, ftp)  # 0.7.0-DEV.2562
```
Comment (ararslan) [inline src/FTPObject.jl]: Probably don't need the version number in a comment, since it doesn't make sense without the `@compat` annotation.

Comment (fchorney) [reply src/FTPObject.jl]: errr I meant to delete that like 3 times, and I kept on forgetting. Thanks for the catch!

Comment (iamed2) [inline src/FTPObject.jl]:
```suggestion
        finalizer(close, ftp)
```

**Pull request merged**